var/spack/repos/builtin/packages/dftbplus/package.py

```diff
@@ -3,85 +3,158 @@
     )
-
     variant(
         "elsi",
         default=False,
         description="Use the ELSI library for large scale systems. "
         "Only has any effect if you build with '+mpi'",
     )
-
+    variant(
+        "gpu",
+        default=False,
+        description="Use the MAGMA library " "for GPU accelerated computation",
+    )
+    variant(
+        "mbd",
+        default=False,
+        when="21.1:",
+        description="Whether DFTB+ should be built with many-body-dispersion support.",
+    )
+    variant("mpi", default=False, description="Whether DFTB+ should support MPI-parallelism.")
+    variant(
+        "openmp",
+        default=True,
+        description="Whether OpenMP thread parallisation should be enabled.",
+    )
+    variant(
+        "plumed",
+        default=False,
+        when="@20.1:",
+        description="Whether metadynamics via the PLUMED2 library should be allowed.",
+    )
+    variant("poisson", default=False, description="Whether the Poisson-solver should be included.")
+    variant(
+        "python",
+        default=False,
+        description="Whether the Python components of DFTB+ should be tested and installed.",
+    )
+    variant(
+        "sdftd3",
+        default=False,
+        when="@21.2:",
+        description="Whether the s-dftd3 library should be included",
+    )
     variant(
         "sockets",
         default=False,
         description="Whether the socket library " "(external control) should be linked",
     )
-
-    variant("arpack", default=False, description="Use ARPACK for excited state DFTB functionality")
-
     variant(
         "transport",
         default=False,
         description="Whether transport via libNEGF should be included. "
         "Only affects parallel build. "
         "(serial version is built without libNEGF/transport)",
     )
+    variant(
+        "tblite",
+        default=False,
+        when="@21.2:",
+        description="Whether xTB support should be included via tblite.",
+    )
+
+    variant("shared", default=False, description="Most often for the Python wrappers.")
 
+    # ONLY for version 19.1
     variant(
         "dftd3",
         default=False,
+        when="@:21.1",
         description="Use DftD3 dispersion library " "(if you need this dispersion model)",
     )
 
-    depends_on("lapack")
-    depends_on("blas")
-    depends_on("scalapack", when="+mpi")
-    depends_on("mpi", when="+mpi")
+    depends_on("cmake@3.16:", type="build")
+    depends_on("ninja@1.10", type="build")
+
+    depends_on("blas", when="-mpi")
+    depends_on("lapack", when="-mpi")
+
+    depends_on("arpack-ng", when="+arpack")
+    depends_on("simple-dftd3", when="+sdftd3")
     depends_on("elsi", when="+elsi")
     depends_on("magma", when="+gpu")
-    depends_on("arpack-ng", when="+arpack")
+    depends_on("mpi", when="+mpi")
+    depends_on("plumed", when="+plumed")
+    depends_on("scalapack", when="+mpi")
+    depends_on("python", when="+python")
```

Comment:
We can make this a little more concrete with specifications from 
- python>3.2 https://github.com/dftbplus/dftbplus/blob/e724a2d5f90bed716337abafbf9cc2884e535cca/INSTALL.rst#L105
- numpy requirement: https://github.com/dftbplus/dftbplus/blob/e724a2d5f90bed716337abafbf9cc2884e535cca/tools/dptools/README#L65
```suggestion
    depends_on("python@3.2:", type=("build", "run"))
    depends_on("py-numpy", type=("build", "run"))  # for tests
```